### PR TITLE
[Pundit] Invert skip_authorizations

### PR DESCRIPTION
**Pull request by malted**
Closes https://github.com/hackclub/hcb/issues/6314

### CODE REVIEW

File: app/controllers/donations_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class DonationsController < ApplicationController
   include SetEvent
   include Rails::Pagination
 
-  skip_after_action :verify_authorized, except: [:start_donation, :make_donation]
+  skip_after_action :verify_authorized, only: [:export, :show, :refund, :qr_code, :finish_donation, :finished]
```
Comment (sampoder) [inline app/controllers/donations_controller.rb]: Looking at this list, we probably don't want to skip verifying that we've authorized the donation for some of these. Certainly export, show, and refund. Ik we used to but we probs want to stop doing that.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -3,7 +3,30 @@
...
+                                               :exchange_login_code,
+                                               :logout,
+                                               :unimpersonate,
+                                               :receipt_report,
+                                               :edit_featurepreviews,
+                                               :edit_security,
+                                               :edit_admin,
+                                               :toggle_sms_auth,
+                                               :complete_sms_auth_verification,
+                                               :start_sms_auth_verification]
```
Comment (sampoder) [inline app/controllers/users_controller.rb]: Similar comment to above, we probably want to start verifying that we authorise some of these.

File: app/controllers/donations_controller.rb (hunk 1 above)
Comment (malted) [reply app/controllers/donations_controller.rb]: Agreed. How about we merge this in to stop the problem from getting worse (this PR changes no current behaviour) and then do an audit pass of all the auth skipped methods after?

Comment (sampoder) [reply app/controllers/donations_controller.rb]: I'd be cool with that

**Review by garyhtou — APPROVED** [review]

**Pull request merged**